graphql_compiler/compiler/ir_lowering_common/location_renaming.py

```diff
@@ -17,7 +23,10 @@ def make_revisit_location_translations(query_metadata_table):
     return location_translations
 
 
-def translate_potential_location(location_translations, potential_location):
+def translate_potential_location(
+    location_translations: Dict[Type[BaseLocation], Type[BaseLocation]],
+    potential_location: Type[BaseLocation],
+) -> Type[BaseLocation]:
```

Comment:
```suggestion
    location_translations: Dict[Location, Location],
    potential_location: LocationT,
) -> LocationT:
```
See below why the `Type` bit is invalid. `LocationT` is because `potential_location` and the return value are guaranteed to be the same type.

It's also not clear in this zoomed-in context, but `location_translations` cannot contain keys or values of type `FoldScopeLocation` — only `Location` ones.